File: sdk/python/src/watcher_sdk/ports/transport.py
# Port: what the use cases need from the network layer.
# Use cases depend on this interface, not on HttpTransport.
# Tests inject FakeTransport — no HTTP calls needed.
from abc import ABC, abstractmethod

from watcher_sdk.domain.event import EventInput


class Transport(ABC):
    """Delivers a batch of events to the gateway."""

    @abstractmethod
    def send(self, events: list[EventInput]) -> None:
        """Send events to the gateway. Raise on unrecoverable failure."""

File: sdk/js/packages/core/src/ports/transport.ts
/**
 * Transport is the port interface for delivering event batches to the gateway.
 * Use cases depend on this interface, not on NodeTransport or BrowserTransport,
 * so tests can inject a FakeTransport without making real HTTP calls.
 */
import type { EventInput } from "../domain/event.js";

export interface Transport {
  /** Send a batch of events. Reject the promise on unrecoverable failure. */
  send(events: EventInput[]): Promise<void>;
}
